cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/DefaultMacroEvaluator.java

```diff
@@ -73,9 +82,15 @@ public String evaluate(String macroFunction, String... arguments) throws Invalid
     if (macroFunction.equals(LOGICAL_START_TIME_FUNCTION_NAME)) {
       return logicalStartTimeMacro.evaluate(logicalStartTime, arguments);
     } else if (macroFunction.equals(SECURE_FUNCTION_NAME)) {
-      // todo lookup secure store, when available
-      return "";
+      if (arguments.length != 1) {
+        throw new InvalidMacroException("Secure store macro function only supports 1 argument.");
+      }
+      try {
+        return new String(secureStore.get(namespace, arguments[0]).get());
+      } catch (IOException e) {
+        throw new InvalidMacroException(e.getMessage());
```

Comment:
Should include the original exception as the cause and with message that is suitable for this context. E.g.

```new InvalidMacroException("Failed to resolve macro '" + macroFunction + "(" + arguments[0] + ")'", e);```